store/models/run.go

```diff
@@ -189,13 +189,23 @@ func (rr RunResult) GetError() error {
 	}
 }
 
-// MergeData merges the existing Data on a RunResult with the given JSON.
-func (rr RunResult) MergeData(j JSON) (RunResult, error) {
-	merged, err := rr.Data.Merge(j)
+// Merge merges the input RunResult with the instance it is called on,
+// preferring the RunResult values passed in, but using the existing values
+// if the input RunResult values are of their respective zero value.
+func (rr RunResult) Merge(in RunResult) (RunResult, error) {
+	merged, err := rr.Data.Merge(in.Data)
 	if err != nil {
-		return rr, fmt.Errorf("TaskRun#Merge merging JSON: %v", err.Error())
+		return in, fmt.Errorf("TaskRun#Merge merging JSON: %v", err.Error())
 	}
-
-	rr.Data = merged
-	return rr, nil
+	in.Data = merged
+	if !in.ErrorMessage.Valid && rr.ErrorMessage.Valid {
+		in.ErrorMessage = rr.ErrorMessage
```

Comment:
If `in` doesn't have an error message, isn't that what should be returned in the final result? In general, I find these assignments of `rr` values when `in` is at the zero value unintuitive.

Reply:
Perhaps we should return an error if you try to merge onto a `RunResult` that's already errored.